llmfoundry/data/finetuning/tasks.py

```diff
@@ -109,37 +109,42 @@ def _is_empty_or_nonexistent(dirpath: str) -> bool:
 
 def _get_role_key(message: Dict[str, str]) -> str:
     role_keys = _ALLOWED_ROLE_KEYS.intersection(message.keys())
-    assert len(role_keys) == 1
+    assert len(
```

Comment:
I missed in the initial PR I guess, but we use `raise` for errors that might actually happen and we want to communicate to the user, and `assert` is just for documenting invariants, satisfying pyright, or in tests. Could you update everywhere to `raise`?